### PR TITLE
Bump ESLint configs and related dependencies

### PR DESCRIPTION
**Pull request by Mrtenz**
This bumps ESLint and configs to the latest version, and fixes linting issues.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -69,5 +103,5 @@ module.exports = {
     },
   ],
 
-  ignorePatterns: ['!.eslintrc.js', '!.prettierrc.js', 'dist/'],
+  ignorePatterns: ['!.eslintrc.js', '!.prettierrc.js', 'dist/', 'docs/'],
```
Comment (mcmire) [inline .eslintrc.js:106]: Good call, maybe we should do this for the module template.

File: src/BIP44CoinTypeNode.ts [hunk 2]
```diff
@@ -147,7 +147,8 @@ export class BIP44CoinTypeNode implements BIP44CoinTypeNodeInterface {
     validateCoinType(coin_type);
     validateCoinTypeNodeDepth(node.depth);
 
-    return new BIP44CoinTypeNode(node, coin_type);
+    // TODO: Make this function not async in a future version.
+    return Promise.resolve(new BIP44CoinTypeNode(node, coin_type));
```
Comment (mcmire) [inline src/BIP44CoinTypeNode.ts:151]: Why was this failing lint? This method should already return a promise, no?

File: src/derivation.ts [hunk 3]
```diff
@@ -169,6 +174,11 @@ export function validatePathSegment(
...
 }
 
+/**
+ * Get the error for a malformed path segment.
+ *
+ * @returns The error.
+ */
 function getMalformedError() {
-  throw new Error('Invalid HD path segment: The path segment is malformed.');
+  return new Error('Invalid HD path segment: The path segment is malformed.');
```
Comment (mcmire) [inline src/derivation.ts:183]: Oooh, good catch!

File: tsconfig.json [hunk 4]
```diff
@@ -9,5 +9,5 @@
     "strict": true,
     "target": "ES2017"
   },
-  "include": ["src"]
+  "include": ["src", "test"]
```
Comment (mcmire) [inline tsconfig.json:12]: With this setting, could test files end up in the build without us knowing?

What do you think about defining two tsconfigs, one for dev and one for building? See: https://github.com/MetaMask/metamask-module-template/blob/main/tsconfig.build.json#L11

**Review by mcmire — COMMENTED** [review]
The latest updates to the ESLint rules are great, looks like we can clean up a lot of stuff. Just had one thing, really.

File: src/BIP44CoinTypeNode.ts (hunk 2 above)
Comment (Mrtenz) [reply src/BIP44CoinTypeNode.ts:151]: `async` functions must use `await` or return a `Promise` in the function. This function previously didn't have any async logic, and it should be changed to not return a `Promise`, but doing so would be a breaking change.

File: tsconfig.json (hunk 4 above)
Comment (Mrtenz) [reply tsconfig.json:12]: Done!

File: .eslintrc.js (hunk 1 above)
Comment (Mrtenz) [reply .eslintrc.js:106]: Looks like this is already done in the module template.

Comment (mcmire) [reply .eslintrc.js:106]: Ah my bad. Cool :)

**Pull request merged**